atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/service/SplitKeyDelegatingTransactionService.java

```diff
@@ -106,15 +118,38 @@ public void close() {
         keyedServices.values().forEach(TransactionService::close);
     }
 
-    private ListenableFuture<Long> getFromDelegate(
-            Map<T, ? extends AsyncTransactionService> keyedTransactionServices, long startTimestamp) {
+    // todo(snanda): This looks not pretty
+    private <Status> ListenableFuture<Status> getFromDelegate(
```

Comment:
Convention for type parameters is typically single uppercase letters, or if we feel we need to be descriptive, then `BlahT` i.e. `StatusT` here?